src/main.go

```diff
@@ -138,13 +151,15 @@ func main() {
 					}
 					chainSet.Register(fmt.Sprintf("%s-monitor", uHash), matrix[hash].MonitorChain)
 				}
+				// If the route is already present in the matrix,
+				// the route is already monitored, we just continue
+				// normal execution.
 			}
 			log.Infoln("initialization time: " + time.Since(p).String())
-			done <- struct{}{}
```

Comment:
If this is removed, how will the reloading caller know that the reload process is completed?

Reply:
You could do a `log.Infoln`

Reply:
I mean to say, the caller means an API. The API needs to send back success or failure. The success is only when the reload success signal is sent. Logging wouldn't help that.

Reply:
Done